src/test/java/org/kohsuke/github/GHPullRequestTest.java

```diff
@@ -421,6 +421,66 @@ public void setLabels() throws Exception {
         assertEquals(label, labels.iterator().next().getName());
     }
 
+    @Test
+    // Requires push access to the test repo to pass
+    public void addLabels() throws Exception {
+        GHPullRequest p = getRepository().createPullRequest("addLabels", "test/stable", "master", "## test");
+        String addedLabel1 = "addLabels_label_name_1";
+        String addedLabel2 = "addLabels_label_name_2";
+        String addedLabel3 = "addLabels_label_name_3";
+
+        p.addLabels(addedLabel1);
+        p.addLabels(addedLabel2, addedLabel3);
+
+        Collection<GHLabel> labels = getRepository().getPullRequest(p.getNumber()).getLabels();
+        assertEquals(3, labels.size());
+        assertThat(labels,
+                containsInAnyOrder(hasProperty("name", equalTo(addedLabel1)),
+                        hasProperty("name", equalTo(addedLabel2)),
+                        hasProperty("name", equalTo(addedLabel3))));
+    }
+
+    @Test
+    // Requires push access to the test repo to pass
+    public void addLabelsConcurrencyIssue() throws Exception {
+        String addedLabel1 = "addLabelsConcurrencyIssue_label_name_1";
+        String addedLabel2 = "addLabelsConcurrencyIssue_label_name_2";
+
+        GHPullRequest p1 = getRepository()
+                .createPullRequest("addLabelsConcurrencyIssue", "test/stable", "master", "## test");
+        p1.getLabels();
+
+        GHPullRequest p2 = getRepository().getPullRequest(p1.getNumber());
+        p2.addLabels(addedLabel2);
+
+        p1.addLabels(addedLabel1);
+
+        Collection<GHLabel> labels = getRepository().getPullRequest(p1.getNumber()).getLabels();
+        assertEquals(2, labels.size());
+        assertThat(labels,
+                containsInAnyOrder(hasProperty("name", equalTo(addedLabel1)),
+                        hasProperty("name", equalTo(addedLabel2))));
+    }
+
+    @Test
+    // Requires push access to the test repo to pass
+    public void removeLabels() throws Exception {
+        GHPullRequest p = getRepository().createPullRequest("removeLabels", "test/stable", "master", "## test");
+        String label1 = "removeLabels_label_name_1";
+        String label2 = "removeLabels_label_name_2";
+        String label3 = "removeLabels_label_name_3";
+        p.setLabels(label1, label2, label3);
+
+        Collection<GHLabel> labels = getRepository().getPullRequest(p.getNumber()).getLabels();
+        assertEquals(3, labels.size());
+
+        p.removeLabels(label2, label3);
+
+        labels = getRepository().getPullRequest(p.getNumber()).getLabels();
+        assertEquals(1, labels.size());
+        assertEquals(label1, labels.iterator().next().getName());
+    }
```

Comment:
Hm, if you the following above this line, you'll get a `GHFileNotFoundException`. 

```
        p.removeLabels(label2);
```

Need to test this scenario.  Also, it seems like this could still result in concurrency issues.  Two clients go to remove the same label and one gets an error while the other doesn't.  Would you rather merge this change with this or work on a solution?

Reply:
if you... what? There's a word missing :)

I don't see how the case you're talking about is a concurrency issue? For sure, if you have several clients removing the same labels at the same time, you will have errors. But that's how the APIs are designed and that's expected in my opinion.

What's not acceptable is the issue that this PR fixes e.g. your use of the API is perfectly fine and you lose data.

Reply:
We probably want to have `removeLabels(...)` match the old method behavior and not throw if any of the label is not present. 
And then add a `removeLabel(String)` that does throw if the specified label isn't present.